Integrations/integration-AzureAD.yml

```diff
@@ -0,0 +1,157 @@
+        4.3 Application ID:  The Application ID (from section 1)
+        4.4 Application Secret: the application secret (from section 1)
+
+configuration:
+- display: Fetch incidents
+  name: isFetch
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Incident type
+  name: incidentType
+  defaultvalue: ""
+  type: 13
+  required: false
+- display: Login URL (e.g. https://login.microsoftonline.com)
+  name: login_url
+  defaultvalue: https://login.microsoftonline.com
+  type: 0
+  required: true
+- display: Tenant Domain
+  name: domain
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Application ID
+  name: client_id
+  defaultvalue: ""
+  type: 0
+  required: false
+- display: Application Secret
+  name: secret
+  defaultvalue: ""
+  type: 4
+  required: false
+script:
+  script: |
+    var sendRequest = function(url, method, headers, body) {
+        var res = http(
+            url,
+            {
+                Method:  method,
+                Headers: {
+                    'Content-Type': ['application/x-www-form-urlencoded'],
+                    'Authorization' : token? [token_type + ' ' + token] : undefined
+                }, //TODO add headers
+                Body : body? encodeToURLQuery(body).substr(1) : undefined
+            },
+            params.insecure,
+            params.proxy
+        );
+        if (res.StatusCode < 200 || res.StatusCode >= 300) {
+            throw 'Request Failed.\nStatus code: ' + res.StatusCode + '.\nBody: ' + JSON.stringify(res) + '.';
+        }
+        return res.Body;
+    };
+
+    var client_id = params.client_id;
+    var client_secret = params.secret;
+    var login_url = params.login_url.replace(/[\/]+$/, '');
+    var tenant_domain = params.domain.replace(/[\/]+$/, '');
+
+    var token;
+    var token_type;
+
+    var getToken = function(){
+        request_url = login_url + '/'+tenant_domain + '/oauth2/v2.0/token';
+        bodyvals = {
+            'client_id': client_id,
+            'client_secret': client_secret,
+            'grant_type': 'client_credentials',
+            'scope': 'https://graph.microsoft.com/.default'
+        };
+        res = sendRequest(request_url, 'POST', {}, bodyvals);
+        try {
+            result = JSON.parse(res);
+        } catch (err) {
+            throw 'Request Failed. \n'+ res;
+        }
+        token = result['access_token'];
+        token_type = result['token_type'];
+        if(!token || !token_type){
+            throw 'Request failed: missing token\n'+ res;
+        }
+    };
+
+    var fetchLoginEvents = function() {
+        var lastRun = getLastRun();
+        var date = new Date();
+        var now = {value : date.toISOString()};
+        if (!lastRun || !lastRun.value) {
+            date.setDate(date.getDate() - 1);
+            lastRun = {value: date.toISOString()};
+        }
+        request_string = 'https://graph.microsoft.com/beta/identityRiskEvents?filter=createdDateTime%20gt%20'+lastRun.value;
```

Comment:
this looks terrible. probably better to use the correct string and the do encode uri to it